### PR TITLE
[Task] Chat Typing Indicator Presentation

### PR DESCRIPTION
**Pull request by jpeng-ms**
## Purpose
Added presentation layer for typing indicator. The avatar view is currently missing due to an issue with MSFAvatarGroup: 

image

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/Presentation/SwiftUI/Chat/ChatViewComponents/TypingParticipantsView.swift [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+//
+
+import SwiftUI
+
+struct TypingParticipantsView: View {
+    @ObservedObject var viewModel: TypingParticipantsViewModel
+
+    var body: some View {
+        HStack {
+            Text(viewModel.typingParticipants ?? "")
```
Comment (jpeng-ms) [inline AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/Presentation/SwiftUI/Chat/ChatViewComponents/TypingParticipantsView.swift:13]: WIP

Comment (jpeng-ms) [reply AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/Presentation/SwiftUI/Chat/ChatViewComponents/TypingParticipantsView.swift:13]: having some issues with MSFAvatarGroup, please let me know if you would like me to resolve it in next PR

**Comment by jpeng-ms** [comment]
having some issues with MSFAvatarGroup, I am planning to resolve it once I got back from FluentUI team, but please let me know if you would like me to address it in this PR instead.

File: AzureCommunicationUI/Podfile [hunk 2]
```diff
@@ -15,6 +15,7 @@ target 'AzureCommunicationUICalling' do
   pod 'MicrosoftFluentUI/Button_ios', '0.3.9'
   pod 'MicrosoftFluentUI/PopupMenu_ios', '0.3.9'
   pod 'MicrosoftFluentUI/ActivityIndicator_ios', '0.3.9'
+  pod 'MicrosoftFluentUI/AvatarGroup_ios', '0.3.9'
```
Comment (jpeng-ms) [inline AzureCommunicationUI/Podfile:18]: if we don't add AvatarGroup_ios to calling, the build will fail in pipeline. Please let me know if there's a better way for this

File: AzureCommunicationUI/sdk/AzureCommunicationUIChat/Tests/Mocking/ChatSDKEventsHandlerMocking.swift [hunk 3]
```diff
@@ -1,23 +0,0 @@
-//
```
Comment (jpeng-ms) [inline AzureCommunicationUI/sdk/AzureCommunicationUIChat/Tests/Mocking/ChatSDKEventsHandlerMocking.swift:1]: resolving this file as discussed with the team, will add it back in the future if we decided to test out sdk layer

File: AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/Presentation/Factories/CompositeViewModelFactory.swift [hunk 4]
```diff
@@ -70,4 +71,10 @@ class CompositeViewModelFactory: CompositeViewModelFactoryProtocol {
         MessageInputViewModel(logger: logger,
                               dispatch: dispatch)
     }
+
+    // MARK: Typing Indicators
```
Comment (palatter) [inline AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/Presentation/Factories/CompositeViewModelFactory.swift:75]: Probably don't need a seperate 'MARK' here

Comment (jpeng-ms) [reply AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/Presentation/Factories/CompositeViewModelFactory.swift:75]: thanks, I am removing it

File: AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/Localization/Localizable.strings [hunk 5]
```diff
@@ -7,3 +7,6 @@
 "AzureCommunicationUIChat.ChatView.TopBarView.WaitingForOthersToJoin"="Waiting for others to join";
 "AzureCommunicationUIChat.ChatView.TopBarView.ChatWith1Person"="Chat with 1 person";
 "AzureCommunicationUIChat.ChatView.TopBarView.ChatWithNPeople"="Chat with %d people"; // %d is for number of people in call
+/* TypingParticipantsView */
+"AzureCommunicationUIChat.ChatView.TypingParticipantsView.AreTyping"="are typing...";
+"AzureCommunicationUIChat.ChatView.TypingParticipantsView.IsTyping"="is typing...";
```
[the remark this answers is not in the document]
Comment (jpeng-ms) [reply AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/Localization/Localizable.strings:12]: thanks, I will update it

**Pull request closed without merging**